tiledb/api/src/filter/ftype.rs

```diff
@@ -23,9 +39,56 @@ pub enum FilterType {
     Delta,
 }
 
-impl FilterType {
-    pub(crate) fn capi_enum(&self) -> ffi::tiledb_filter_type_t {
-        match *self {
+impl ToStringCore for FilterType {
```

Comment:
I'm not entirely sure that `Core` is the best suffix here. Though I haven't got anything better to suggest so I can't really ask for a change...

Reply:
I'm going back and forth on this one a bit.

When working within Rust only, the core "to string" function is a pain because the thing it prints out does not match the name of the enum variant.  This is a pain if you want to copy/paste code from a Debug trait output into code, which is something I have done a lot for converting proptest output into reproducers.

But the core to/from string functions are also the canonical representations which I assume are used by all the other language bindings.  So we want them for inter-op.

Maybe the answer is that Debug should do the Rust thing (i.e. `#[derive(Debug)]`) but we should use the core string functions for `Display` and `FromStr`?

Otherwise these traits aren't really used seriously so adding them to the list of ignored C API functions.  IIRC these are only present for Datatype and FilterOption.

Reply:
FWIW, I was only concerned about "core" vs something like "internal" since its what the library uses in a few places. It just felt like core being a word we use to refer to libtiledb.so and fiends might not carry the same meaning for someone not familiar with how we use that.

In terms of whether to even expose these, I don't know of any other bindings that use them to actually transfer meaning around vs "show me a debug description of this thing" like the dump functions for larger classes like Array and Query and so on. And given that we get that basically for free from Rust and `derive(Debug)`, I'd be fine just blanket ignoring all of the to_str/from_str functions right along with the dump functions.

Reply:
Great, I'll move them to ignored then!

Reply:
Removed the traits and impls and added the functions to `ignored.rs`